### PR TITLE
mgr/cephadm: track and expose daemon ports; dynamically assign rgw ports

### PR DESCRIPTION
**Pull request by liewegas**
- push ports (and ip) down into unit.meta
- include in 'ls' result (filling in if not already defined in unit.meta)
- report in 'ceph orch ps' result
```
NAME                 HOST  STATUS        REFRESHED  AGE   PORTS   VERSION                IMAGE NAME                                                                                          IMAGE ID      CONTAINER ID  
crash.dael           dael  running (2h)  10m ago    2h    -       17.0.0-1275-g5e197a21  docker.io/ceph/daemon-base@sha256:8a9d01990f2601ea3111d84a18ba84a9043275cde1e48956a74b5a12e5e850b3  5c2a636e05b4  f2aa107e9edc  
rgw.r.z.dael.hwdftp  dael  error         10m ago    112m  *:8123  <unknown>              docker.io/ceph/daemon-base@sha256:8a9d01990f2601ea3111d84a18ba84a9043275cde1e48956a74b5a12e5e850b3  <unknown>     <unknown>     
```
- dynamically assign ports for rgw, allowing more than one instance per host

Fixes: https://tracker.ceph.com/issues/48822

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/python-common/ceph/deployment/service_spec.py [hunk 1]
```diff
@@ -567,6 +567,12 @@ def service_name(self) -> str:
             n += '.' + self.service_id
         return n
 
+    def get_ports(self) -> List[int]:
+        return []
```
Comment (sebastian-philipp) [inline src/python-common/ceph/deployment/service_spec.py:571]: The ports are per-daemon, but `ServiceSpec` is per Service. If you add this **only** to this class, it's going to be painful to support different ports for different daemons. 

Also, we have default ports for different service types. and the MGR is super special.

File: src/python-common/ceph/deployment/service_spec.py [hunk 2]
```diff
@@ -567,6 +567,12 @@ def service_name(self) -> str:
...
+
+    def get_ip(self) -> Optional[str]:
+        return None
```
Comment (sebastian-philipp) [inline src/python-common/ceph/deployment/service_spec.py:574]: `get_ip()` can't live in ServiceSpec. This is a property of a host and not of a service spec.

**Comment by liewegas** [comment]
https://pulpito.ceph.com/sage-2021-03-12_18:50:58-rados:cephadm-wip-sage-testing-2021-03-12-1116-distro-basic-smithi

(upgrade tests are stuck for unrelated reasons; will retest once the other fixes merge)

**Comment by liewegas** [comment]
https://pulpito.ceph.com/sage-2021-03-16_05:26:44-rados:cephadm-wip-sage3-testing-2021-03-15-2317-distro-basic-smithi/

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**